src/components/common/YourCompany/OfficeAddresses.vue

```diff
@@ -542,18 +546,34 @@ export default class OfficeAddresses extends Mixins(CommonMixin) {
         )
       }
     }
+
+    if (this.isChangeFiling || this.isConversionFiling) {
+      if (this.getOfficeAddresses?.businessOffice) {
+        this.mailingAddress = { ...this.getOfficeAddresses.businessOffice.mailingAddress }
+        this.deliveryAddress = { ...this.getOfficeAddresses.businessOffice.deliveryAddress }
+
+        // compare addresses to set the "inherit mailing" flag
+        // ignore Address Type since it's different
+        // ignore Address Country Description since it's not always present
+        this.inheritMailingAddress = isSame(
+          this.getOfficeAddresses.businessOffice.mailingAddress,
+          this.getOfficeAddresses.businessOffice.deliveryAddress,
+          ['addressType', 'addressCountryDescription', 'id']
+        )
+      }
+    }
   }
 
   /**
    * When "same as (registry) mailing address" checkbox is changed,
    * sets the Registered Delivery Address to the Registered Mailing Address.
    */
-  private setDeliveryAddressToMailingAddress (): void {
+  protected setDeliveryAddressToMailingAddress (): void {
     if (this.inheritMailingAddress) {
       this.deliveryAddress = { ...this.mailingAddress, addressType: 'delivery', id: this.deliveryAddress.id }
     } else {
       // clear to default
-      this.deliveryAddress = { ...this.defaultAddress, addressType: 'delivery' }
+      this.deliveryAddress = { ...this.defaultAddress, addressType: 'delivery', id: this.deliveryAddress.id }
```

Comment:
Vysakh, this is the bug! When the checkbox is clicked and the address is cleared here, it needed to retain the original ID.

The other code that something similar (for records office), on line 596-597, was already correct.